### ISSUE TITLE
Variable 'e' is no longer parsed properly

### ISSUE THREAD

**Issue by keyneom** [pieterderycke/Jace#38]
A change was recently made to add support for scientific notation. I believe in doing so we lost the ability to use the built-in variable `e`.

Attempts to use any formula using the variable `e` as previously done now fail. e.g.

```
var engine = new CalculationEngine(CultureInfo.InvariantCulture, ExecutionMode.Compiled);
var result = engine.Calculate("e");
```

When using more complex functions (`1 * e`) you receive a rather unclear error message:

```
There is a syntax issue for the operation \"*\" at position 2. The number of arguments does not match with what is expected.
```

Reverting the following line change seems to avoid the issue, but I am sure will also prevent the scientific notation support from functioning properly:
https://github.com/pieterderycke/Jace/blob/25b8d70f903be6a27a2513f3a87db936d4c20981/Jace/Tokenizer/TokenReader.cs#L224

**Comment by keyneom** [comment]
@pieterderycke Was this an intended change? There are some performance changes recently released I am anxious to try out but the above issue is preventing us from doing so.

**Comment by pieterderycke** [comment]
@keyneom Thanks for letting me know. I will look to fix the bug and to release a bug fix release

**Comment by pieterderycke** [comment]
I have released a Jace 0.9.2 to fix this bug

**Comment by keyneom** [comment]
Amazingly fast!! Thank you so much!!
